### ISSUE TITLE
Fix how `original_size` and `addr` are treated for Fake AIL Blocks

### ISSUE THREAD

**Issue by mahaloz** [angr/angr#4572]
### Background
After debugging some code that flowed through `to_ail_supergraph`, it became clear that this line is causing problems:
https://github.com/angr/angr/blob/b0396c9fd3d72c0b1ab47fd0956654cef1366a60/angr/analyses/decompiler/utils.py#L301

For completely normal AIL nodes (ones converted from Vex and untouched), this is a sane check. However, in the case of AIL nodes that have been _created_, this check fails. Any of the `DUPLICATING` optimizations, can create new nodes:
https://github.com/angr/angr/blob/b0396c9fd3d72c0b1ab47fd0956654cef1366a60/angr/analyses/decompiler/optimization_passes/cross_jump_reverter.py#L99

At first, I thought you course just set `original_size=None`, and then do a special check in `to_ail_supergraph`, but this causes crashes elsewhere.

In summary, angr is not positioned to handle AIL nodes that have either fake addresses or fake sizes. We need to fix this eventually. 

### The Fix
1. Introduce a new property to nodes `is_fake`, meaning the node was created 
2. Go through all places that use `.addr` or `.original_size` and update them to no longer trust these values
